typing/typedecl.ml

```diff
@@ -1198,6 +1198,10 @@ let transl_type_decl env rec_flag sdecl_list =
   (* Compute the final environment with variance and immediacy *)
   let final_env = add_types_to_env decls shapes env in
   (* Check re-exportation *)
+  let check_abbrev env name_sdecl (id, slot) =
```

Comment:
Rather than shadowing the name of the global function, you could inline this local function inside `List.iter2` or use a distinct name.